### PR TITLE
Added promotion based on User Group

### PR DESCRIPTION
**Pull request by antonioperic**
Added promotion based on User Group. Also little update for backend user show (added list of groups)

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml [hunk 1]
```diff
@@ -81,15 +81,6 @@ sylius_product:
             form:
                 default: Sylius\Bundle\CoreBundle\Form\Type\ProductType
 
-sylius_archetype:
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: This should be reverted probably...

**Comment by stloyd** [comment]
Seems like your rebase gone wrong in few places...

File: src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml (hunk 1 above)
Comment (antonioperic) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: This exist in master as I can see

Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: And you delete it in this PR. :)

Comment (antonioperic) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: fixing... :)

**Comment by antonioperic** [comment]
@pjedrzejewski updated

**Comment by antonioperic** [comment]
is now ok? cc @stloyd @pjedrzejewski

File: src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml (hunk 1 above)
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: You need to revert this change, it's wrong to remove that configuration.

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -206,13 +207,6 @@
             <tag name="kernel.event_listener" event="sylius.user.post_create" method="processUser" />
             <tag name="kernel.event_listener" event="sylius.user.post_update" method="processUser" />
         </service>
-        <service id="sylius.listener.user_delete" class="%sylius.listener.user_delete.class%">
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml]: Seems wrong.

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -59,7 +58,9 @@
         <parameter key="sylius.promotion_rule_checker.user_loyalty.class">Sylius\Component\Core\Promotion\Checker\UserLoyaltyRuleChecker</parameter>
         <parameter key="sylius.promotion_rule_checker.shipping_country.class">Sylius\Component\Core\Promotion\Checker\ShippingCountryRuleChecker</parameter>
         <parameter key="sylius.promotion_rule_checker.taxonomy.class">Sylius\Component\Core\Promotion\Checker\TaxonomyRuleChecker</parameter>
+        <parameter key="sylius.promotion_rule_checker.user_group.class">Sylius\Component\Core\Promotion\Checker\UserGroupRuleChecker</parameter>
         <parameter key="sylius.promotion_rule_checker.contains_product.class">Sylius\Component\Core\Promotion\Checker\ContainsProductRuleChecker</parameter>
+        <parameter key="sylius.promotion_rule_checker.user_group.class">Sylius\Component\Core\Promotion\Checker\UserGroupRuleChecker</parameter>
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml]: Duplicated line from above.

File: src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml [hunk 4]
```diff
@@ -1,13 +1,9 @@
-# This file is part of the Sylius package.
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml]: Should be reverted.

File: src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml [hunk 5]
```diff
@@ -175,8 +171,8 @@ sylius:
             index: Coupons
         attribute:
             index: Attributes
-        archetype:
-            index: archetypes
+        prototype:
+            index: Prototypes
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml]: Should be reverted.

File: src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml [hunk 6]
```diff
@@ -598,7 +594,6 @@ sylius:
         contain_variants: Contain Variants
         create: Create product
         create_header: New product
-        archetype: Archetype
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml]: Should be reverted.

File: src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml [hunk 7]
```diff
@@ -700,19 +695,17 @@ sylius:
...
-        create: Create archetype
-        create_header: New archetype
-        index_header: Product archetypes
-        code: code
+    prototype:
+        build: Build prototype
+        build_header: Building product from prototype
+        create: Create prototype
+        create_header: New prototype
+        index_header: Product prototypes
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml]: Should be reverted.

File: src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml [hunk 8]
```diff
@@ -700,19 +695,17 @@ sylius:
...
         name: name
-        no_results: There are no archetypes defined.
+        no_results: There are no prototypes defined.
         options: options
-        attributes: attributes
-        parent: parent
-        update_header: Editing archetype
+        attributes:
+        update_header: Editing prototype
```
Comment (stloyd) [inline src/Sylius/Bundle/WebBundle/Resources/translations/messages.en.yml]: Should be reverted.

**Comment by antonioperic** [comment]
@stloyd thnx fill fix things

**Comment by antonioperic** [comment]
@stloyd now good?

**Comment by antonioperic** [comment]
And squashed

**Comment by stloyd** [comment]
@antonioperic Looks good now :+1:

**Comment by pjedrzejewski** [comment]
Thank you Antonio! :+1:

**Comment by antonioperic** [comment]
Finally, that rebase went pretty wrong :)
